### ISSUE TITLE
Mark 5b time setting does not also set frame_nr

### ISSUE THREAD

**Issue by mhvk** [mhvk/baseband#158]
(Might affect mark4 as well)
```
from astropy.time import Time
from baseband.mark5b import mark5b
mark5b.Mark5BHeader.fromvalues(time=Time('2014-06-13T05:30:01'), frame_nr=89)
<Mark5BHeader sync_pattern: 0xabaddeed,
              user: 0,
              internal_tvg: False,
              frame_nr: 89,
              bcd_jday: 0x821,
              bcd_seconds: 0x19801,
              bcd_fraction: 0x0,
              crc: 0x975d>
```
Here, one has an inconsistent header, since at an integer second, `frame_nr` should be 0.

**Comment by cczhu** [comment]
Curiously, `Mark5Bheader.get_time` still uses `framerate` rather than `sample_rate`...

Mark 4 is not affected, since 16 Msps / track is the maximum sampling rate [(design doc Sec. 2)](https://www.haystack.mit.edu/tech/vlbi/mark5/docs/230.3.pdf) and so the timespan between frames is within the precision of the timestamp.

**Comment by cczhu** [comment]
I don't think there is a `frame_nr` for Mark 4.  Such a property doesn't exist in Baseband, and I don't think one's mentioned in the [Mark 4 specs](https://www.haystack.mit.edu/tech/vlbi/mark5/docs/230.3.pdf) either.

**Comment by mhvk** [comment]
Ah, my mistake!

**Comment by mhvk** [comment]
I was thinking a bit more about how to address it, and one problem is that `sample_rate` is not enough to determine the frame number from the time -- we do not have access to `bps` and `nchan`, so do not know `samples_per_frame`! Indeed, this must be why `framerate` is used.

**Comment by cczhu** [comment]
Solved by #173.
